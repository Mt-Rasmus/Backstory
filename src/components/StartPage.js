
import React from 'react';
import { Link } from 'react-router-dom';
import InstructionsModal from './InstructionsModal';

export default class StartPage extends React.Component {
   constructor(props) {
      super(props);
      this.state = {
         showModal: false
      };
   }

   toggleShowModal = () => {
      this.setState({ showModal: !this.state.showModal })
   }

   render() {
      return (
         <div className="start-options-layout">
            <div className="start-options-layout__box">
               <Link to="/stories">
                  <button className="button button-big button--standard">
                     Stories
                  </button>
               </Link>
               <button 
                  className="button button-big button--standard" 
                  onClick={this.toggleShowModal}>How to play</button>      
               <div>
                  <InstructionsModal 
                     showModal={this.state.showModal} 
                     handleToggleModal={this.toggleShowModal} 
                  />           
               </div>           
            </div>
         </div>
      )
   }
}
