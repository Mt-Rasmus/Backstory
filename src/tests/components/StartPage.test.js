
import React from 'react';
import StartPage from '../../components/StartPage';
import { shallow } from 'enzyme';

test('Should render StartPage correctly', () => {
   const wrapper = shallow(<StartPage />);
   expect(wrapper).toMatchSnapshot();
});
