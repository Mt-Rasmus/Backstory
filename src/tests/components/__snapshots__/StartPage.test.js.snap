// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render StartPage correctly 1`] = `
<div
  className="start-options-layout"
>
  <div
    className="start-options-layout__box"
  >
    <[object Object]
      to="/stories"
    >
      <button
        className="button button-big button--standard"
      >
        Stories
      </button>
    </[object Object]>
    <button
      className="button button-big button--standard"
      onClick={[Function]}
    >
      How to play
    </button>
    <div>
      <InstructionsModal
        handleToggleModal={[Function]}
        showModal={false}
      />
    </div>
  </div>
</div>
`;
